akka-tools-persistence/src/main/scala/no/nextgentel/oss/akkatools/persistence/EnhancedPersistentActor.scala

```diff
@@ -80,27 +124,20 @@ abstract class EnhancedPersistentActor[E:ClassTag, Ex <: Exception : ClassTag]
   // we will not perform the fix described above again
   protected def getDMGeneratingVersion = 0
 
-  // Info about all already processed (successfully) inbound DMs.
-  // If we see one of these again, we know that it is a resending caused by the sender not
-  // getting the DM-confirm.
-  // So instead of trying to process them - and fail since we've already processed them,
-  // we can 'ignore' them...
-  // But we cannot just ignore them:
-  // If the sender never got our DMReceived, we must send it again..
-  // And since the impl might not have just sent confirm, but used the DM with a new payload to send
-  // a cmd back to the sender via the DM-confirm, we must facilitate for that to also work again..
-  // You know.... idempotent cmds :)
-  private var processedDMs = Set[ProcessedDMEvent]() // DMs without payload
+
+  //Changes to these two has to be backwards compatible
+  case class FullSnapshotState(userData : Any, localState : InternalState)

```

Comment:
If this one is going to be serialized using json, you have to do it annotate userData like payload in DurableMessage